internal/commands/app/create_inputs_test.go

```diff
@@ -372,111 +373,659 @@ func TestAppCreateInputsResolveDirectory(t *testing.T) {
+		}
+
+		dummyClusters := []string{"test-cluster-dummy-1", "test-cluster-dummy-2"}
+		inputs := createInputs{
+			newAppInputs:        newAppInputs{Name: testApp.Name, Project: testApp.GroupID},
+			Clusters:            []string{"test-cluster-1", dummyClusters[0], dummyClusters[1]},
+			ClusterServiceNames: []string{"mongodb-atlas"},
+		}
+
+		doneCh := make(chan (struct{}))
+		go func() {
+			defer close(doneCh)
+			console.ExpectString(fmt.Sprintf("Note: The following data sources were not linked because they could not be found: %s", strings.Join(dummyClusters[:], ", ")))
+			console.ExpectEOF()
+		}()
+
+		cmd := &CommandCreate{inputs}
+		assert.Nil(t, cmd.Handler(profile, ui, cli.Clients{Realm: rc, Atlas: ac}))
+
+		ds, _, err := inputs.resolveClusters(ui, ac, "123")
+		assert.Nil(t, err)
+
+		assert.Equal(t, []dataSourceCluster{
+			{
+				Name: "mongodb-atlas",
+				Type: realm.ServiceTypeCluster,
+				Config: configCluster{
+					ClusterName:         "test-cluster-1",
+					ReadPreference:      "primary",
+					WireProtocolEnabled: false,
+				},
+			},
+		}, ds)
+	})
+
+	t.Run("should prompt user for confirmation if clusters are not found", func(t *testing.T) {
+		testApp := realm.App{
+			ID:          primitive.NewObjectID().Hex(),
+			GroupID:     primitive.NewObjectID().Hex(),
+			ClientAppID: "test-app-abcde",
+			Name:        "test-app",
+		}
+
+		rc := mock.RealmClient{}
+		rc.ImportFn = func(groupID, appID string, appData interface{}) error {
+			return nil
+		}
+		rc.TemplatesFn = func() ([]realm.Template, error) {
+			return []realm.Template{}, nil
+		}
+
+		ac := mock.AtlasClient{}
+		ac.ClustersFn = func(groupID string) ([]atlas.Cluster, error) {
+			return []atlas.Cluster{{ID: "789", Name: "test-cluster-1"}}, nil
+		}
+		dummyClusters := []string{"test-cluster-dummy-1", "test-cluster-dummy-2"}
+		inputs := createInputs{
+			newAppInputs:        newAppInputs{Name: testApp.Name, Project: testApp.GroupID},
+			Clusters:            []string{"test-cluster-1", dummyClusters[0], dummyClusters[1]},
+			ClusterServiceNames: []string{"mongodb-atlas"},
+		}
+
+		for _, tc := range []struct {
+			description string
+			response    string
+			appCreated  bool
+		}{
+			{
+				description: "and quit if not confirmed",
+				response:    "no",
+				appCreated:  false,
+			},
+			{
+				description: "and continue to create app if confirmed",
+				response:    "yes",
+				appCreated:  true,
+			},
+		} {
+			t.Run(tc.description, func(t *testing.T) {
+				profile, teardown := mock.NewProfileFromTmpDir(t, "app_create_test")
+				defer teardown()
+
+				_, console, _, ui, err := mock.NewVT10XConsole()
+				assert.Nil(t, err)
+				defer console.Close()
+
+				doneCh := make(chan (struct{}))
+				go func() {
+					defer close(doneCh)
+					console.ExpectString(fmt.Sprintf("Note: The following data sources were not linked because they could not be found: %s", strings.Join(dummyClusters[:], ", ")))
```

Comment:
same nit here about `[:]` and/or just writing the expected string as is (without `fmt.Sprintf`)